docs_src/threat-models/secret-store/threat_model.md

```diff
@@ -0,0 +1,353 @@
+
+* Consul data store is assumed to be non-confidential and thus there is no threat. Vault data is encrypted prior to be passed to Consul for storage.
+
+#### (j2) Loss of integrity or availability of Consul data store at-rest by file system tampering or malicious deletion.
+
+* Container/Snap provided file system protections.
+
+#### (j3) Loss of availability of Consul data store at runtime due to intentional service crash.
+
+* In Docker-based environments, Consul can be configured to automatically restart on failure.
+* Threat may be further mitigated by running Consul in High Availability mode (not done in reference implementation).
+
+#### (k1) Loss of confidentiality of PKI CA at-rest by file system inspection/monitoring.
+
+* Container/Snap provided file system protections.
+* Secure deletion of CA private key after PKI generation.
+
+#### (k2) Loss of integrity of PKI CA by malicious replacement.
+
+* Container/Snap provided file system protections.
+
+#### (k3) Loss of availability of PKI CA (public certificate) by malicious deletion.
+
+* Container/Snap provided file system protections.
+
+#### (l1) Loss of confidentiality of PKI intermediate at-rest by file system inspection/monitoring.
+
+* Container/Snap provided file system protections.
+* Secure deletion of CA intermediate private key after PKI generation.
+
+#### (l2) Loss of integrity of PKI intermediate by malicious replacement.
+
+* Identical to threat (k3): CA would have to be maliciously replaced as well.
+
+#### (l3) Loss of availability of PKI intermediate (public certificate) by malicious deletion.
+
+* Container/Snap provided file system protections.
+
+#### (m1) Loss of confidentiality of PKI leaf at-rest by file system inspection/monitoring.
+
+* Container/Snap provided file system protections.
+* Note that server TLS private keys must be delivered to services unencrypted due to limitations of dependent services.
+
+#### (m2) Loss of integrity of PKI leaf by malicious replacement.
+
+* Identical to threat (k3/l3): CA or intermediate would have to be maliciously replaced as well.
+
+#### (m3) Loss of availability of PKI leaf by malicious deletion.
+
+* Container/Snap provided file system protections.
+
+#### (p) Disclosure, tampering, or deletion of secrets through /proc/mem or ptrace() by malicous or compromised microservice
+
+- Container/Snap provided memory protections.
+
+#### (q) Lost of confidentiality of input key material (IKM)
+
+- IKM is secured by vendor-defined hardware-mechanism.
+- IKM is passed to key derivation function via IPC pipe (stdout).
+
+
+## Known vulnerabilites
+
+**Service location tampering vulnerability:**
+Services use configuration information to connect to the secret store and other services.
+The configuration information in all current EdgeX releases is provided
+by Consul and bootstrapped by a `configuration.toml` file in each service.
+Consul is currently run as a public service (no encryption, authentication, or access control)
```

Comment:
Aren't we now securing consul to some extent?  Do we need to address to what degree here?

Reply:
Removed this section.  This has been addressed.  This document was written well over a year ago.